dask/array/tests/test_array_core.py

```diff
@@ -3057,6 +3041,7 @@ def test_warn_bad_rechunking():
     assert '20' in record[0].message.args[0]
 
 
+@pytest.mark.xfail(reason='atop fusion doesnt respect this, which is ok')
```

Comment:
Marking this so changing/removing this xfailed test isn't forgotten.

Reply:
I wasn't planning to fix this. The `fuse_keys` functionality was only there to ensure that we didn't optimize out a key that we cared about.  We handle that with other functionality now.  Should I remove the test entirely, or should we work to retain the old behavior?

Reply:
I'd remove the test entirely.